07.ls_object/file_display.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+class FileDisplay
+  COLUMN_COUNT = 3
+
+  def initialize
+    @options = ARGV.getopts('alr').transform_keys(&:to_sym)
+    @file_details = fetch_file_details
+  end
+
+  def show_formatted_file_details
+    @options[:l] ? long_format : short_format
+  end
+
+  private
+
+  def fetch_file_details
+    flags = @options[:a] ? File::FNM_DOTMATCH : 0
+    file_names = Dir.glob('*', flags)
+    sorted_file_names = @options[:r] ? file_names.reverse : file_names
+    sorted_file_names.map { |file_name| FileDetail.new(file_name) }
+  end
+
+  def long_format
+    blocks_total = @file_details.map(&:blocks).sum
+    puts "total #{blocks_total}"
+
+    @file_details.each { |file_detail| puts format_file_detail(file_detail) }
+  end
+
+  def format_file_detail(file_detail)
+    max_size_map = build_max_size_map(@file_details)
+
+    [
+      file_detail.fetch_file_mode_strings,
+      format("%#{max_size_map[:nlink] + 1}s", file_detail.nlink),
+      format("%-#{max_size_map[:user_name] + 1}s", file_detail.user_name),
```

Comment:
```suggestion
      file_detail.nlink.to_s.rjust(max_size_map[:nlink] + 1),
      file_detail.user_name.ljust(max_size_map[:user_name] + 1),
```

個人的にはljustやrjustを使った方が可読性が高い（意図が伝わりやすい）んじゃないかなーと思います

Reply:
おっしゃる通り、`ljust`、`rjust`を使ったほうが可読性が高そうですね✨
「"% n s"」、「"%- n s"」を使うよりも`ljust`、`rjust`を使うほうが、コードを見た時に左右どちら寄りになるかをぱっと理解できそうです。
こちら修正しました。